src/java_bytecode/java_local_variable_table.cpp

```diff
@@ -720,8 +722,24 @@ void java_bytecode_convert_methodt::setup_local_variables(
   for(const auto &v : m.local_variable_table)
     vars_with_holes.push_back({v, is_parameter(v), {}});
 
-  // Merge variable records:
-  find_initialisers(vars_with_holes, amap, dominator_analysis);
+  // Merge variable records. See documentation of in
+  // `find_initialisers_for_slot` for more details. If the strategy employed
+  // there fails with an exception, we just ignore the LVT for this method, no
+  // variable is generated in `this->variables[]` (because we return here and
+  // dont let the for loop below to execute), and as a result the method
+  // this->variable() will be forced to create new `anonlocal::` variables, as
+  // the only source of variable names for that method is `this->variables[]`.
+  try
+  {
+    find_initialisers(vars_with_holes, amap, dominator_analysis);
+  }
+  catch(const char *message)
+  {
+    error() << "Bytecode -> codet translation error: " << message << eom
```

Comment:
I would say it's a serious warning, or recoverable error: if we get here it's because the LVT looks much alien to the bytecode -> codet translation algorithms, and other thinks could break.